rero_ils/modules/deduplications/api.py

```diff
@@ -0,0 +1,523 @@
+            return None
+        return (
+            1
+            if cls.normalize(data_field[0]["_text"][0]["value"])
+            == cls.normalize(c_field[0]["_text"][0]["value"])
+            else 0
+        )
+
+    @classmethod
+    def get_edition_statement_score(cls, data, candidate):
+        """Get the edition statement score.
+
+        :returns: the edit distance of the edition statement field.
+        """
+        data_field = data.get("editionStatement", [{"_text": [{"value": ""}]}])
+        c_field = candidate.get("editionStatement", [{"_text": [{"value": ""}]}])
+        if not data.get("editionStatement") and not candidate.get("editionStatement"):
+            return None
+        return (
+            1
+            if cls.normalize(data_field[0]["_text"][0]["value"])
+            == cls.normalize(c_field[0]["_text"][0]["value"])
+            else 0
+        )
+
+    @classmethod
+    def get_doc_type_score(cls, data, candidate):
+        """Get the document type score.
+
+        :returns: 1 if at least one main type matched.
+        """
+        return (
+            1.0
+            if (
+                {t["main_type"] for t in data["type"]}
+                & {t["main_type"] for t in candidate["type"]}
+            )
+            else 0
+        )
+
+    @classmethod
+    def get_publication_date_score(cls, data, candidate):
+        """Get the document type score.
+
+        :returns: 1 if both exist and are equal matched, 0 otherwise.
+        """
+        publication_date = data.get("sort_date_old")
+        c_publication_date = candidate.get("sort_date_old")
+        if publication_date and c_publication_date:
+            return 1 if publication_date == c_publication_date else 0
+        else:
+            return 0 if bool(publication_date) != bool(c_publication_date) else 1
+
+    @classmethod
+    def get_provision_activity_score(cls, data, candidate):
+        """Get the provision activity score.
+
+        :returns: 1 if both exist and are equal matched, 0 otherwise.
+        """
+        data_field = data["provisionActivity"][0]["_text"][0]["value"]
+        c_field = candidate["provisionActivity"][0]["_text"][0]["value"]
+        return cls.edit_distance(data_field, c_field)
+
+    @classmethod
+    def get_extent_score(cls, data, candidate):
+        """Get the provision activity score.
+
+        :returns: 1 if both exist and are equal matched, 0 otherwise.
+        """
+        data_field = set(re.findall(r"\d+", data.get("extent", "")))
+        c_field = set(re.findall(r"\d+", candidate.get("extent", "")))
+        if data_field and c_field:
+            return len(data_field & c_field) / max(len(data_field), len(c_field))
+        else:
+            return 0 if bool(data_field) != bool(c_field) else 1
+
+    @classmethod
+    def get_responsibility_score(cls, data, candidate):
+        """Get the document type score.
```

Comment:
Compare and score the similarity of the first responsibilityStatement's `_text` in both documents.